File: Creational/Abstract Factory/Java/ConcreteFactory1.java

public class ConcreteFactory1 implements AbstractFactory {

	@Override
	public AbstractProductA createProductA() {
		return new ConcreteProductA_1();
	}

	@Override
	public AbstractProductB createProductB() {
		return new ConcreteProductB_1();
	}

	@Override
	public AbstractProductC createProductC() {
		return new ConcreteProductC_1();
	}
	
}

File: Creational/Abstract Factory/Kotlin/ConcreteFactory1.kt
class ConcreteFactory1: AbstractFactory {

    override fun createProductA(): AbstractProductA {
        return ConcreteProductA1()
    }

    override fun createProductB(): AbstractProductB {
        return ConcreteProductB1()
    }

    override fun createProductC(): AbstractProductC {
        return ConcreteProductC1()
    }

}